src/timely.rs

```diff
@@ -66,7 +100,7 @@ where
 
     /// Is a given epoch closed based on this frontier?
     fn is_closed(&self, epoch: &T) -> bool {
-        self.simplify().iter().all(|frontier| *epoch < *frontier)
+        !self.simplify().is_some_and(|frontier| *epoch >= frontier)
```

Comment:
This raises our minimum supported version to `1.70.0`, so we need to update CI and docker images. I think we should start keeping track of this in a `rust-toolchain.toml` file to make it explicit when we need to do that, but you'll need to do it in this PR too to make the CI pass